APE14.rst

```diff
@@ -242,13 +248,18 @@ low-level API recommends:
             n_world scalars or arrays as input in units given by ``world_axis_units``.
             Returns n_pixel scalars or arrays. Note that pixel coordinates are
             assumed to be 0 at the center of the first pixel in each dimension.
+            to be 0 at the center of the first pixel in each dimension. If a
+            world coordinate does not have a matching pixel coordinate, NaN can
+            be returned.  The coordinates should be returned in the ``(x, y)``
```

Comment:
I am understanding this correctly that this is *not* numpy order (numpy being rows, columns for 2D right?)? If so I am infinitely :-1: on this, this is a Python API and should be doing things in C order everywhere.

Reply:
I think of this as a WCS API and as such it is Cartesian coordinates order, i.e. (x, y, z). I believe this is what is common for other packages as well so there's some expectation of this order.

In practical terms, since `astropy.wcs` and `gwcs` use Cartesian order too (not sure about LSST's software), this means the implementation does not have to rearrange the order of the inputs. It would be awkward for the high level interface and the low level object to work with a different order of inputs.
Although I see the appeal of using numpy order.

Reply:
Does gwcs actually enforce this? I have been constructing all my gwcses in numpy order to avoid this disparity.

I really really dislike the fact that astropy WCS axes order does not match the order of the numpy array I think it leads to a really bad user experience, given this API is Python all the way down I really believe that the best thing to do is be consistent with the ordering of the arrays you are presumably working with when you make your WCS.

For just a little bit of context this is what the `repr` for a `NDCube` looks like:
```
NDCube
---------------------
WCS Keywords

Number of WCS axes: 3
CTYPE : 'HPLN-TAN'  'HPLT-TAN'  'TIME'  
CRVAL : 0.0  0.0  0.0  
CRPIX : 2060.610107  2040.26001  1.0  
PC1_1 PC1_2 PC1_3  : 0.99999711341582  0.0024027401094365  0.0  
PC2_1 PC2_2 PC2_3  : -0.0024027401094365  0.99999711341582  0.0  
PC3_1 PC3_2 PC3_3  : 0.0  0.0  1.0  
CDELT : 0.00016669694444444  0.00016669694444444  12.00000000000188  
NAXIS : 0  0
---------------------
Length of NDCube: [1703. 4096. 4096.] pix
Axis Types of NDCube: ('time', 'custom:pos.helioprojective.lat', 'custom:pos.helioprojective.lon')
```

I really do not like the way the dimensions of the array and the WCS have a mismatch. I have been telling @danryanirsh and everyone else who will listen that gWCS and the APE 14 API will remove this glaring inconsistency, and I am very disappointed that it might not!

Reply:
As discussed on slack, we could agree to change it, I think this clearly just needs more discussion. But the APE as-is was under-specific about this.